local/network.go

```diff
@@ -325,6 +326,37 @@ func NewDefaultConfig(binaryPath string) network.Config {
 	return config
 }
 
+// NewDefaultConfigNNodes creates a new default network config, with an arbitrary number of nodes
+func NewDefaultConfigNNodes(binaryPath string, numNodes uint32) (network.Config, error) {
+	netConfig := NewDefaultConfig(binaryPath)
+	if int(numNodes) > len(netConfig.NodeConfigs) {
+		toAdd := int(numNodes) - len(netConfig.NodeConfigs)
+		refNodeConfig := netConfig.NodeConfigs[0]
+		for i := 0; i < toAdd; i++ {
+			nodeConfig := refNodeConfig
+			stakingCert, stakingKey, err := staking.NewCertAndKeyBytes()
+			if err != nil {
+				return netConfig, fmt.Errorf("couldn't generate staking Cert/Key: %w", err)
+			}
+			nodeConfig.StakingKey = string(stakingKey)
+			nodeConfig.StakingCert = string(stakingCert)
+			// replace api port in refNodeConfig.ConfigFile
+			apiPort, err := getFreePort()
```

Comment:
don't we need to also look for a free P2P port?

Reply:
No. Reason is that the reference node config sets only API port. I need to replace that conf with a new port in flags. Regarding the P2P port, as is not defined in any conf, it will be automatically generated.